### PR TITLE
add opcodes to sync messages

### PR DESCRIPTION
**Pull request by adambabik**
## Changes

1. Two opcodes were introduced `p2pSyncRequestCode` and `p2pSyncResponseCode`,
2. MailServer interface got extended with `SyncMail(*Peer, SyncMailRequest) error`.

There are **no** breaking changes.

Relates to https://github.com/status-im/status-go/issues/1290.

### CODE REVIEW

File: whisperv6/whisper.go [hunk 1]
```diff
@@ -452,6 +452,35 @@ func (whisper *Whisper) SendHistoricMessageResponse(peer *Peer, payload []byte)
...
+}
+
+// SendSyncResponse sends a response to a Mail Server with a slice of envelopes.
+func (whisper *Whisper) SendSyncResponse(p *Peer, data SyncResponse) error {
+	size, r, err := rlp.EncodeToReader(data)
+	if err != nil {
+		return err
+	}
+
+	return p.ws.WriteMsg(p2p.Msg{Code: p2pSyncResponseCode, Size: uint32(size), Payload: r})
```
Comment (dshulyak) [inline whisperv6/whisper.go]: same

File: whisperv6/whisper.go [hunk 2]
```diff
@@ -452,6 +452,35 @@ func (whisper *Whisper) SendHistoricMessageResponse(peer *Peer, payload []byte)
...
+	if err != nil {
+		return err
+	}
+
+	size, r, err := rlp.EncodeToReader(req)
+	if err != nil {
+		return err
+	}
+
+	return p.ws.WriteMsg(p2p.Msg{Code: p2pSyncRequestCode, Size: uint32(size), Payload: r})
```
Comment (dshulyak) [inline whisperv6/whisper.go]: i think you can use p2p.Send here, it will do same thing as added by you

**Comment by dshulyak** [comment]
i didnt notice at first that there are not tests. will you add them?)

**Comment by adambabik** [comment]
@dshulyak added missing unit tests :)

**Pull request merged**